### PR TITLE
Update copyright and license headers

### LINKED ISSUE #6073: Copyright in files
The repo is under MIT license.

Currently many files in the repo contain Microsoft Copyrights. New files haven't copyrights at all.
All files in CoreFX repo has MIT copyright banner. This is expected for the repo too (including distributed files).

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Based on standard practices, we need to have a copyright and license notice at the top of each source file.  Removed existing copyrights and updated/added copyright notices for .h, .cpp, .cs, .ps1, and .psm1 files.

Updated module manifests for consistency to have Author = "PowerShell" and Company = "Microsoft Corporation".  Removed multiple linebreaks.

Separate PR coming to update contribution document for new source files: https://github.com/PowerShell/PowerShell/pull/6140

Manually reviewed each change.

Fix https://github.com/PowerShell/PowerShell/issues/6073

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [X] Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: src/System.Management.Automation/engine/parser/FusionAssemblyIdentity.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (lzybkr) [inline src/System.Management.Automation/engine/parser/FusionAssemblyIdentity.cs]: This file is copied from Roslyn which is uses Apache, not MIT.

File: src/System.Management.Automation/engine/parser/GlobalAssemblyCache.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (lzybkr) [inline src/System.Management.Automation/engine/parser/GlobalAssemblyCache.cs]: This file is copied from Roslyn which is uses Apache, not MIT.

**Review by lzybkr — DISMISSED** [review]
I did a spot check and found 2 files that are licensed with Apache that you changed. You need to re-review how you made these changes to ensure there aren't others.

File: src/System.Management.Automation/engine/parser/FusionAssemblyIdentity.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/parser/FusionAssemblyIdentity.cs]: I'll put the big banner version of the Apache copyright here

File: src/System.Management.Automation/engine/parser/GlobalAssemblyCache.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/parser/GlobalAssemblyCache.cs]: I'll put the big banner version of the Apache copyright here

**Comment by SteveL-MSFT** [comment]
@lzybkr I saw the big banner versions that indicate Apache License and left those, but must have just automatically replaced these single line versions.  Did a search of master and saw 39 hits with Apache License.  Searched this branch and also 39 hits (now).

File: build.psm1 [hunk 3]
```diff
@@ -2434,6 +2431,53 @@ function Clear-PSRepo
     }
 }
 
+# Install PowerShell modules from a git Repo such as PSL-Pester
+function Restore-GitModule
+{
+    [CmdletBinding()]
+    param(
```
Comment (TravisEz13) [inline build.psm1]: added back removed code

Comment (SteveL-MSFT) [reply build.psm1]: will remove

**Comment by TravisEz13** [comment]
I see changes in SessionStateContainer.cs as well
image

**Comment by TravisEz13** [comment]
code deleted out of /Get-ChildItem.Tests.ps1

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments

**Comment by SteveL-MSFT** [comment]
@TravisEz13 I painstakingly went through each diff individually using the GitHub view and don't see the changes you show in those two files.  I do see some changes in PSReadLine sources which I'm going to revert.

**Review by TravisEz13 — APPROVED** [review]

**Comment by iSazonov** [comment]
Should we add meta-test to keep consistency?

**Comment by iSazonov** [comment]
I found that a blank line is absent in some cpp and h files after the header.

**Comment by SteveL-MSFT** [comment]
@iSazonov having a meta-test sounds like a good idea, but we should have that added outside this PR.  I'll add the whitespace between the header and first line of code.

**Comment by SteveL-MSFT** [comment]
@lzybkr are you ok with the changes and verification of the changes?

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**